Jdaviz Subset Theory Docs (#207)
spacetelescope/jdaviz: docs/dev/infrastructure.rst

Note that those libraries themselve depend on the wider scientific Python ecosystem, so this is by no means a comprehensive list of the libraries jdaviz depends on.  But these are the primary "top-level" libraries, meaning it is these that most users are likely to focus on to parallel or extend their jdaviz workflows.

Selections
==========

This section explains the working theory behind the selection tool and was inspired by the the introduction of two methods to the SpecViz helper:
`specviz.get_spectra()` and `specviz.get_spectral_regions()`

> Data loaded in are imported into jdaviz and immediately converted into a SpecUtils SpectralRegion. These are meant to be the spectral analog to the Astropy Region (for spatial WCS). These regions are returned by the `specviz.get_spectra()` method.

Reviewer: ```suggestion
Data loaded in are imported into jdaviz and immediately converted into a `specutils.SpectralRegion`. These are a spectral analog to the Astropy `regions` (which instead focuses on spatial regions and their associated WCS). These spectral regions are returned by the `specviz.get_spectra()` method.
```

> The selection tool allows the user to specify a specific range on the graph. This is defined by the underlying Glue Library upon which jdaviz relies on as a "Glue Subset." Thus throughout this section, we will refer to these user defined ranges as "subsets." Effectively, the selection tool defines a mask that can be thought of as "definition" of which data is and is not included in the subset. Upon extraction via `specviz.get_spectral_regions()`, the method will return a new SpecUtils SpectralRegion object that applies that mask atop of the proper region (data) displayed, and realizes the subset the user defined in jdaviz. 

Reviewer: ```suggestion
The selection tool allows the user to specify a specific range on the graph. This is defined by the underlying Glue Library upon which jdaviz relies on as a "Glue Subset." Thus throughout this section, we will refer to these user defined ranges as "subsets." Effectively, the selection tool defines a mask that can be thought of as "definition" of which data is and is not included in the subset. Upon extraction via `specviz.get_spectral_regions()`, the method will return a new `specutils.SpectralRegion` object that applies that mask atop of the proper region (data) displayed, and realizes the subset the user defined in jdaviz. 
```
